### PR TITLE
Feature 13085 self report edit

### LINKED ISSUE #13085: Add an edit/delete/archive functionality for Self Reporting messages (UI) [3]
### Feature Description

Backend is already available
=> Edit form needs to be added

### Added Value/Benefit

To be able to edit/archive-dearchive/delete-restore self reported messages

### Acceptance Criteria

- [x] corresponding rights are added 
- [ ]
- [ ]

### Implementation Details

_No response_

### Mockups

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by leventegal-she**
Fixes #13085

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=13100

File: sormas-backend/src/main/resources/sql/sormas_schema.sql [hunk 1]
```diff
@@ -13100,4 +13100,32 @@ ALTER TABLE contact_history ADD COLUMN casereferencenumber text;
 
 INSERT INTO schema_version (version_number, comment) VALUES (546, 'Extend the contacts with Case Reference Number & add case reference number to cases #13067');
 
+-- 2024-05-07 #13085 Add an edit/delete/archive functionality for Self Reporting messages (UI)
+INSERT INTO userroles_userrights (userrole_id, userright)
+SELECT id, 'SELF_REPORT_ARCIVE'
```
Comment (carina29) [inline sormas-backend/src/main/resources/sql/sormas_schema.sql:13105]: Typo in user right

**Review by carina29 — CHANGES_REQUESTED** [review]
There are some conflicts in strings.properties
and SelfReportGrid.java

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=13100

**Review by carina29 — COMMENTED** [review]
Approved PR.

**Pull request merged**